### PR TITLE
update TagBot

### PR DESCRIPTION
**Pull request by jishnub**
Switch to the [suggested workflow](https://discourse.julialang.org/t/ann-required-updates-to-tagbot-yml/49249) to work around github disabling actions on inactive repos.

### CODE REVIEW

File: .github/workflows/TagBot.yml [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+  workflow_dispatch:
 jobs:
   TagBot:
+    if: github.event_name == 'workflow_dispatch' || github.actor == 'JuliaTagBot'
     runs-on: ubuntu-latest
     steps:
       - uses: JuliaRegistries/TagBot@v1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+          ssh: ${{ secrets.DOCUMENTER_KEY }}
```
Comment (johnnychen94) [inline .github/workflows/TagBot.yml:15]: This is specifically needed when unit test CI is run on github action to work around the limitation that "one github action cannot trigger another one". MappedArrays is still using travis so we don't need this.

Most importantly, I believe for this repo, `DOCUMENTER_KEY` key isn't set.

Comment (jishnub) [reply .github/workflows/TagBot.yml:15]: You're right, but do you plan to move away from travis at some point?

Comment (johnnychen94) [reply .github/workflows/TagBot.yml:15]: I believe we should but given that this package isn't actively developed, this is of low priority. I don't have write access to this repo so we should leave @timholy to do so.

**Comment by timholy** [comment]
I added the key.

**Pull request merged**